src/middlewared/middlewared/plugins/interface/lag.py

```diff
@@ -11,30 +11,67 @@ class Config:
     @private
     def lag_setup(self, lagg, members, disable_capabilities, parent_interfaces, sync_interface_opts):
         name = lagg['lagg_interface']['int_interface']
-        self.logger.info('Setting up {}'.format(name))
+        self.logger.info('Setting up %s', name)
 
         try:
             iface = netif.get_interface(name)
         except KeyError:
             iface = None
         else:
-            first_port = next(iter(iface.ports))
+            first_port = next(iter(iface.ports), None)
             if first_port is None or first_port[0] != members[0]['lagg_physnic']:
-                self.logger.info('Destroying existing %s as its first port has changed', name)
+                self.logger.info('Destroying %s because its first port has changed', name)
                 netif.destroy_interface(name)
                 iface = None
 
         if iface is None:
             netif.create_interface(name)
             iface = netif.get_interface(name)
 
+        # FIXME: this will fail on SCALE since we don't even have
+        # a "disable_capabilities" method
+        """
         if disable_capabilities:
             self.middleware.call_sync('interface.disable_capabilities', name)
+        """
 
+        info = {'protocol': None, 'xmit_hash_policy': None, 'lacpdu_rate': None}
         protocol = getattr(netif.AggregationProtocol, lagg['lagg_protocol'].upper())
         if iface.protocol != protocol:
-            self.logger.info('{}: changing protocol to {}'.format(name, protocol))
-            iface.protocol = protocol
+            info['protocol'] = protocol
+
+        xmit_hash = lagg['lagg_xmit_hash_policy']
+        if iface.xmit_hash_policy != xmit_hash:
+            info['xmit_hash_policy'] = xmit_hash
+
+        lacpdu_rate = lagg['lagg_lacpdu_rate']
+        if iface.lacpdu_rate != lacpdu_rate:
+            info['lacpdu_rate'] = lacpdu_rate
+
+        if any(i is not None for i in info.values()):
+            # means one of the lagg options changed or is being
+            # setup for the first time so we have to down the
+            # interface before performing any of the actions
+            iface.down()
+
+            if info['protocol'] is not None:
+                # we _always_ have to start with the protocol
+                # information first since it deletes members
+                # (if any) of the current lagg and then changes
+                # the protocol
+                self.logger.info('Changing protocol on "%s" to %s', name, info['protocol'].name)
```

Comment:
Let's please use `%r` instead of `"%s"`

Reply:
Why? I'm printing the value of the enum object, which is a string?

Reply:
I recommended it because `%r` uses `repr` which handles the quotes automatically and we don't have to add them manually each time.

Reply:
Well that's convenient.